src/ansys/fluent/core/search.py

```diff
@@ -281,10 +282,13 @@ def _write_api_tree_file(api_tree_data: dict, api_object_names: list):
                 if sys.version_info[0] < 3
                 else wn.synsets(name, lang="eng")
             )
-            synset_names = []
+            synset_names = set()
             for api_object_name_synset in api_object_name_synsets:
-                synset_names.append(api_object_name_synset.name().split(".")[0])
-            all_api_object_name_synsets[name] = synset_names
+                synset_names.add(api_object_name_synset.name().split(".")[0])
+            if synset_names:
+                if name in synset_names:
+                    synset_names.remove(name)
```

Comment:
```suggestion
                synset_names.discard(name)
```

Reply:
Right. It  does not raise an exception. Done. Thank you.